shoreline.go

```diff
@@ -178,17 +179,17 @@ func main() {
 	 * Serve it up
 	 */
 	logger.Print("creating http server")
-	server := common.NewServer(&http.Server{
+	server := http.Server{
 		Addr:    config.Service.GetPort(),
 		Handler: rtr,
-	})
+	}
 
 	logger.Print("starting http server")
 	if err := server.ListenAndServe(); err != nil {
 		logger.Fatal(err)
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
This context is used by the kafka consumer - I believe what you have here will shutdown the server after 5 seconds, as it will unblock the consumer, allowing the process to terminate:
```
// blocks until context is canceled right after server.Close()
if err := consumer.Start(ctx); err != nil {		if err := consumer.Start(ctx); err != nil 
```

Instead, you should create a new background context for graceful shutdown:
```
sig := <-signals
shutdownCtx, shutdownCancel := context.WithCancel(context.Background())
defer shutdownCancel()
if err := server.Shutdown(shutdownCtx); err != nil {
...
```